### PR TITLE
Add support for event mapping based on a base class.

### PR DESCRIPTION
**Pull request by d-shtonda**
When handling events, if the mapping for the base class of event is configured, it will be invoked as well as a handler for a concrete event type, if exists.

### CODE REVIEW

File: Src/LiquidProjections/EventMap.cs [hunk 1]
```diff
@@ -73,6 +76,22 @@ internal void AddFilter(Func<object, TContext, Task<bool>> filter)
             }
         }
 
+        private List<Handler> GetHandlersForType(Type eventType)
+        {
+            if (mappings.TryGetValue(eventType, out var handlers))
```
Comment (dennisdoomen) [inline Src/LiquidProjections/EventMap.cs]: This implies that if there's an exact match for the more specific event, it won't even look at the more generic one. In the old VisionSuite denormalizers, we would invoke both handlers.

Comment (d-shtonda) [reply Src/LiquidProjections/EventMap.cs]: Yeah, I agree, invoking both handlers would be better

**Comment by d-shtonda** [comment]
@dennisdoomen, updated to run all handlers (base class and specific ones).
Also fixed randomly failing spec.

**Comment by dennisdoomen** [comment]
Did you leave it on WIP on purpose?

**Comment by d-shtonda** [comment]
It should be good to go.

**Pull request merged**